lib/module.js

```diff
@@ -170,7 +166,13 @@ Module._findPath = function(request, paths) {
 
     if (filename) {
       // Warn once if '.' resolved outside the module dir
-      if (request === '.' && i > 0) noopDeprecateRequireDot();
+      if (request === '.' && i > 0) {
+        warned = internalUtil.printDeprecationMessage(
+          '(node) warning: require(\'.\') resolved outside the package ' +
+          'directory. This functionality is deprecated and will be removed ' +
+          'soon.', warned);
+      }
```

Comment:
Not if you somehow can detect an internal caller. Maybe a second method is necessary? (util.printDeprecationMessage vs. internalUtil.printDeprecationMessage)